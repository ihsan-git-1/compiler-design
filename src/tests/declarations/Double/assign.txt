void main(){

double x=4.1, t=2.2, g, h,z=1.3;
z=3.4;
}