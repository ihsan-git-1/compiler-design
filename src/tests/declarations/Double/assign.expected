
*********** AST ************

Line : (1) Top Tree Declaration  Child Count =  1 Type = Top Tree
 ,  Line : (1) Dart Variable Declaration:  parent: TopTreeDeclaration Child Count =  1 Type = object
Line : (1) Function name: main functionType: void parameters: null functionBody: Line : (1) FunctionBody{statementList=Line : (3) Dart Variable Declaration:  parent: Statement Child Count =  1 Type = object
Line : (3) Double Declaration Line: Number of declarations on this line: 3Number of assignments on this line: 3 parent: Variable Child Count =  10 Type = variable declaration line (type double)
 Line : (4) Double Assignment: z parent: Assignment Child Count =  4 Type = variable assignment (type double)
value: Line : (4) Add Double Expression: 3.4 parent DoubleAssignment Child Count =  1 Type = addition expression (type double)
Line : (4) Number Double Class: num= 3.4 parent: DoubleAssignment Child Count =  1 Type = value (type double)
 col=0 type='Function' childCount=0 parent='null'} parent: 'null Child Count =  5 Type = function

*********** SYMBOL TABLE ************

Global Scope (0)
Function Scope main (1) Parent Is Global Scope (0)

Identifier x, Type double, Value : 4.1 , Scope Function Scope main (1)
Identifier t, Type double, Value : 2.2 , Scope Function Scope main (1)
Identifier g, Type double, Value : 0.0 , Scope Function Scope main (1)
Identifier h, Type double, Value : 0.0 , Scope Function Scope main (1)
Identifier z, Type double, Value : 1.3 , Scope Function Scope main (1)
Identifier z, Type double, Value : 3.4 Scope Function Scope main (1)
